adding sil_p_geez

rik-shaw opened the pull request.

rik-shaw commented:
First attempt at adding a sil_p_geez keyboard.

mcdurdin commented on sil_p_geez/sil_p_geez.kpj:
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="utf-8"?>
+<KeymanDeveloperProject>
+  <Options>
+    <BuildPath></BuildPath>
```
Set this to $PROJECTPATH\build

mcdurdin commented on sil_p_geez/sil_p_geez.kpj:
```diff
@@ -0,0 +1,30 @@
+<KeymanDeveloperProject>
+  <Options>
+    <BuildPath></BuildPath>
+    <CompilerWarningsAsErrors>False</CompilerWarningsAsErrors>
```
Please set this to True unless your keyboard has unresolveable warnings.

mcdurdin commented on sil_p_geez/sil_p_geez.kpj:
```diff
@@ -0,0 +1,30 @@
+    <WarnDeprecatedCode>True</WarnDeprecatedCode>
+  </Options>
+  <Files>
+    <File>
```
Please add the source\sil-p-geez.kmn file to the project

mcdurdin commented on sil_p_geez/source/sil-p-geez.kmn:
```diff
@@ -0,0 +1,345 @@
+
+c ----------------------------------------------------------------------------------
+c This new store (24-Sept-2007) is a fix for the problem of fast typing reordering keystrokes when using Keyman with TSF
+store(chars) 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%^&*()_+-=[]\{}|;:",./<>?' "'`~"
```
This should no longer be required -- this issue was resolved a while back.

mcdurdin commented on sil_p_geez/source/sil-p-geez.kps:
```diff
@@ -0,0 +1,2 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Package><System><KeymanDeveloperVersion>9.0.523.0</KeymanDeveloperVersion><FileVersion>7.0</FileVersion></System><Options><ExecuteProgram></ExecuteProgram><ReadMeFile>sil-p-geez.htm</ReadMeFile><MSIFileName></MSIFileName><MSIOptions></MSIOptions></Options><Buttons><Button><ButtonType>install</ButtonType><Caption>&amp;Install</Caption><URL></URL><Left>0</Left><Top>0</Top></Button><Button><ButtonT ...
```
You won't need the .kmn referenced in the .kps file, but you will need the .kmx (from the build/ folder)

rik-shaw commented:
@mcdurdin I have renamed to "sil_ethiopic_power_g" and hopefully have everything right now.  So the "official name" is "SIL Ethiopic Power G" to comply, but in my .htm file I refer to it as "SIL Ethiopic Power-G" so that "G" isn't all lonely.

Interestingly, it seems that Keyman Developer is wiping out the <BuildPath> and resetting the <CompilerWarningsAsErrors> back to "False".  I fixed them up manually.

Also, the reason my .kmn had a bad encoding from before was as you guessed: I had developed the .kmn in Linux (without a UTF-8 Preamble), but then when imported into Keyman Developer it got it mixed up and when I saved it there it changed the encoding.  But if I start an empty file with Keyman Developer it does save it as UTF-8 correctly.

mcdurdin commented:
I made three changes when I accepted this:

1. Removed commas from the ethnologuecode system store contents
2. Fixed up the path of the .kmx in the .kps file to point to the build folder
3. Removed the 'desktop' build target as this one currently causes a web build which is not currently supported with this keyboard
